source/framework/vx_exemplar.c

```diff
@@ -252,6 +255,18 @@ static vx_reference ownCreatePyramidFromExemplar(
     pmd = vxCreatePyramid(context, levels, scale, width, height,
             format);
 
+    if ((vx_enum)VX_SUCCESS == vxGetStatus((vx_reference)pmd))
+    {
+        vx_uint32 i;
+        vx_status status = (vx_enum)VX_SUCCESS;
+        for (i = 0; (i < levels) && ((vx_enum)VX_SUCCESS == status); ++i)
+        {
+            if (exemplar->img[i]->base.supplementary_data)
+            {
+                status = vxSetSupplementaryUserDataObject(&pmd->img[i]->base, exemplar->img[i]->base.supplementary_data);
```

Comment:
the pyramid object is made of different images, and we must attach each supplementary data to reach sub image.
I added a comment.

Reply:
I need to check the supplementary data extension itself if we need to set the supplementary on the object array level. I don't think so because the objarray is only a container, only the objects contained inside have supplementary data.

Reply:
Thanks.  Please let us know the results of your findings

Reply:
@lcweaver 
from the extension:
"Object Arrays
An object array (the container) may also have supplementary data, distinct from the data of the elements of the array. ...."

Reply:
Thanks.  In that case, is it the case that we will need similar logic added to the object array exemplar function?

I would think so given that if the exemplar was created with different supplementary data on the object array vs the element of the object array, then the resulting objects would also need to be set in a similar manner

Reply:
@raphacan My point is that from my reading of the logic above is that when you create a vx_pyramid from exemplar, the individual pyramid levels (i.e., vx_image's) of the newly created pyramid will have supplementary data matching the supplementary data of the corresponding pyramid level of the exemplar.  This looks to be distinct from the supplementary data of the pyramid object itself as per the newly added logic in the tivxCreateReferenceFromExemplar function.  I am in agreement with these changes, as an application could have theoretically set different supplementary data for each level of the pyramid.

In the same way and as per the spec detail you mentioned (i.e., "An object array (the container) may also have supplementary data, distinct from the data of the elements of the array"), when you create an object array from an exemplar, you will need to set the supplementary data of each element of the object array corresponding to the same element index of the exemplar.

Reply:
Per discussion in 12/13 call, @raphacan  will follow up with Steve on the details.  General alignment is that the pyramid levels will retain the supplementary data of the "mother" pyramid, thus resulting in a change of the above code.

@lcweaver will review the test cases and propose the handling of object array from exemplar

Reply:
@raphacan From reviewing the test cases (particularly lines 1018-1061, linked below), it seems like it is at least theoretically possible for a pyramid to have been created with each levels having different supplementary data than the "mother".  I am in agreement still with the details from our meeting indicating that this ideally should not happen.  However, this is not enforced by way of an error in the implementation.  I think that would be preferable if we want to enforce pyramids to not have different supplementary data per level.  As we mentioned on the call, it would also be good to get information from Steve on how this was conceived.

Similarly, on the topic of object arrays, from the same snippet of test cases, it seems like the object array OpenVX object could have different supplementary data than the elements of the array.  Given this observation from the test cases, it looks like more logic should be added to the object array function in vx_exemplar.c (i.e., ownCreateObjectArrayFromExemplar) to handle the supplementary data setting of each level of the object array.

https://github.com/TexasInstruments/tiovx/pull/8/files#diff-ad4d493cc959235f699f7411e793c004804123d74872cc8cb52d584c5f368497R1018

Reply:
@lcweaver 
I did the change for the pyramid: When it is created from exemplar, the "mother" supplementary will be used to set the supplementary of each level.
for the obj_array, I can do the code modification to have the same mechanism, but I don't know how to create a obj_array from an example obj_array. Do you have such a test ?

Reply:
@raphacan Thanks.  From review, I see that we do not have a direct mechanism for this, but I will create a new example of this at least by the end of the week if not sooner

Reply:
@raphacan @lcweaver An example of creating an object array from a different exemplar object array can be seen in this test case: https://github.com/TexasInstruments/tiovx/blob/main/conformance_tests/test_tiovx/test_objarray.c#L26

Reply:
@brockTI thank you for pointing to this example, I used it to extend our tests.
@lcweaver I modified the code to set all the children with the supplementary of the mother (if existing) if creating a object array from exemplar.

Reply:
This looks good, thank you